### DISCUSSION TITLE
Managing scheduled scripts

### DISCUSSION THREAD

**Discussion by lannonbr** [johnlindquist/kit#317] [Q&A]
A variety of my scheduled scripts reach out to machines on my local network (Google Home, raspberry pi, etc), and when I take my machine off my home network to the office or a coffee shop, I don't want scripts to try reaching out to devices I know are not reachable at that moment.

With that, I'm thinking what a good solution would be to manage all of these particular scripts and turning them off when I leave my house. I would rather not write some geolocation logic, but rather a manually run script or something in the app itself to disable these scripts until I turn them back on.

As well, I wonder if it would be a useful to make a "do not disturb" where you can set a timer of 1 hour, 4 hours, 1 day, etc to turn these scheduled scripts off.

**Comment by johnlindquist** [comment]
I'm imagining a solution where scripts could have conditions. Maybe the Mac address of the router you're connected to?

// Condition: mac === '01:02:03:0a:0b:0c'
// Condition: wifi === false

We can grab plenty of info about the current network. Does that seem inline with your thoughts?

**Reply by lannonbr** [reply]
Yeah that would definitely work

**Reply by rvisharma** [reply]
I use something similar, this gives me connected WIFI SSID with which i can detect whether i'm at home or not

```js
const ssid = await $`networksetup -getairportnetwork en0 | cut -d ' ' -f 4`
```
